src/main/java/fr/maxlego08/essentials/module/modules/SanctionModule.java

```diff
@@ -400,4 +407,63 @@ public void seen(CommandSender sender, String ip) {
 
         message(sender, Message.COMMAND_SEEN_IP_LINE, "%ip%", ip, "%players%", userDTOS.stream().map(user -> getMessage(Message.COMMAND_SEEN_IP_INFO, "%name%", user.name())).collect(Collectors.joining(",")));
     }
+
+    public void freeze(CommandSender sender, UUID uuid, String userName) {
+
+        if (isProtected(userName)) {
+            message(sender, Message.COMMAND_SANCTION_ERROR);
+            return;
+        }
+
+        IStorage iStorage = this.plugin.getStorageManager().getStorage();
+        User user = iStorage.getUser(uuid);
+        if (user == null) {
+            message(sender, Message.PLAYER_NOT_FOUND, "%player%", userName);
+            return;
+        }
+
+        user.setFrozen(!user.isFrozen());
+
+        Sanction sanction = Sanction.freeze(uuid, getSenderUniqueId(sender));
+        iStorage.insertSanction(sanction, index -> {
+            sanction.setId(index);
+            iStorage.updateUserFrozen(uuid, user.isFrozen());
+        });
+        if (user.isFrozen()) {
+            message(sender, Message.COMMAND_FREEZE_SUCCESS, "%player%", userName);
+            this.plugin.getEssentialsServer().sendMessage(uuid, Message.MESSAGE_FREEZE);
+
+            user.getPlayer().setAllowFlight(true);
+            user.getPlayer().setFlying(true);
+            user.getPlayer().setFlySpeed(0f);
+            this.plugin.getScheduler().teleportAsync(user.getPlayer(), user.getPlayer().getLocation().add(0, 0.1, 0));
+        } else {
+            user.getPlayer().setAllowFlight(false);
+            user.getPlayer().setFlying(false);
+            message(sender, Message.COMMAND_UN_FREEZE_SUCCESS, "%player%", userName);
+            this.plugin.getEssentialsServer().sendMessage(uuid, Message.MESSAGE_UN_FREEZE);
+        }
+    }
+
+    @EventHandler
+    public void onJoin(PlayerJoinEvent event) {
+        IStorage iStorage = this.plugin.getStorageManager().getStorage();
+        User user = iStorage.getUser(event.getPlayer().getUniqueId());
```

Comment:
You must use the getUser method directly in class EssentialsPlugin, there is also a getUser method in class ZModule